sdk/src/azure/core/az_mqtt5_rpc_client_hfsm.c

```diff
@@ -0,0 +1,841 @@
+    case AZ_EVENT_MQTT5_CONNECTION_CLOSE_REQ:
+    case AZ_MQTT5_EVENT_DISCONNECT_RSP:
+    case AZ_MQTT5_EVENT_UNSUBACK_RSP:
+    case AZ_MQTT5_EVENT_RPC_CLIENT_UNSUB_REQ:
+    case AZ_MQTT5_EVENT_PUB_RECV_IND:
+      break;
+
+    default:
+      ret = AZ_HFSM_RETURN_HANDLE_BY_SUPERSTATE;
+      break;
+  }
+
+  return ret;
+}
+
+/**
+ * @brief If the incoming pub is of the correct topic, this transitions to the ready state if needed
+ * and passes the event for the ready state to handle
+ */
+AZ_INLINE az_result send_to_ready_if_topic_matches(
+    az_mqtt5_rpc_client_policy* this_policy,
+    az_event event,
+    az_event_policy_handler source_state)
+{
+  az_mqtt5_recv_data* recv_data = (az_mqtt5_recv_data*)event.data;
+
+  // Ensure pub is of the right topic
+  if (az_span_topic_matches_sub(
+          this_policy->_internal.rpc_client->_internal.subscription_topic, recv_data->topic))
+  {
+    // transition states if requested
+    if (source_state != NULL)
+    {
+      _az_RETURN_IF_FAILED(
+          _az_hfsm_transition_peer(&this_policy->_internal.rpc_client_hfsm, source_state, ready));
+      // notify application that it can now send invoke requests
+      // if ((az_event_policy*)this_policy->inbound_policy != NULL)
+      // {
+      // az_event_policy_send_inbound_event((az_event_policy*)this_policy, (az_event){.type =
+      // AZ_MQTT5_EVENT_RPC_CLIENT_READY_IND, .data = this_policy->_internal.rpc_client});
+      // }
+      _az_RETURN_IF_FAILED(_az_mqtt5_connection_api_callback(
+          this_policy->_internal.connection,
+          (az_event){ .type = AZ_MQTT5_EVENT_RPC_CLIENT_READY_IND,
+                      .data = this_policy->_internal.rpc_client }));
+
+      // pass event to Ready to be processed there
+      _az_RETURN_IF_FAILED(_az_hfsm_send_event(&this_policy->_internal.rpc_client_hfsm, event));
+    }
+  }
+  return AZ_OK;
+}
+
+/**
+ * @brief Transitions to idle if needed and sends an unsubscribe request
+ */
+AZ_INLINE az_result
+unsubscribe(az_mqtt5_rpc_client_policy* this_policy, az_event_policy_handler source_state)
+{
+  // transition states if requested
+  if (source_state != NULL)
+  {
+    _az_RETURN_IF_FAILED(
+        _az_hfsm_transition_peer(&this_policy->_internal.rpc_client_hfsm, source_state, idle));
+  }
+
+  // Send unsubscribe
+  az_mqtt5_unsub_data unsubscription_data
+      = { .topic_filter = this_policy->_internal.rpc_client->_internal.subscription_topic };
+
+  _az_RETURN_IF_FAILED(az_event_policy_send_outbound_event(
+      (az_event_policy*)this_policy,
+      (az_event){ .type = AZ_MQTT5_EVENT_UNSUB_REQ, .data = &unsubscription_data }));
+
+  return AZ_OK;
+}
+
+static az_result idle(az_event_policy* me, az_event event)
+{
+  az_result ret = AZ_OK;
+  az_mqtt5_rpc_client_policy* this_policy = (az_mqtt5_rpc_client_policy*)me;
+
+  if (_az_LOG_SHOULD_WRITE(event.type))
+  {
+    // printf("%s ", az_span_ptr(this_policy->_internal.rpc_client->_internal.command_name));
```

Comment:
Remove?

Reply:
These were to help discriminate different client policy logs, but I'll remove these for now since there's probably a better solution moving forward anyways.